frontend/internal-packages/agent/src/chat/workflow/shared/langGraphUtils.ts

```diff
@@ -29,7 +28,7 @@ export const DEFAULT_RECURSION_LIMIT = 20
  */
 export const createAnnotations = () => {
   return Annotation.Root({
-    messages: Annotation<BaseMessage[]>,
+    ...MessagesAnnotation.spec,
```

Comment:
I have switched to MessagesAnnotation provided by LangChain. It is still `messages: Annotation<BaseMessage[]>`, but the update method is different and is concated by reducer.

ref: https://langchain-ai.github.io/langgraphjs/concepts/low_level/#messagesannotation

In other words, the `messages: [..... state.messages, newMessage]` is now `messages: [newMessage]`.
This is implicit and confusing, so I implemented it myself, but I had to conform to it in order to use ToolNode.